model/host/db.go

```diff
@@ -567,6 +567,20 @@ func FindOneId(id string) (*Host, error) {
 	return FindOne(ById(id))
 }
 
+func FindOneByIdOrTag(id string) (*Host, error) {
+	query := db.Query(bson.M{
+		"$or": []bson.M{
+			bson.M{TagKey: id},
+			bson.M{IdKey: id},
+		},
+	})
+	host, err := FindOne(query) // try to find by tag
```

Comment:
`host.FindOne` can return `nil, nil` if there is no host found, so check `host == nil` after checking the error.

Reply:
I wanted this function to have the same behavior as FindOneId--it's essentially a FindOneId with an additional field considered for Id, and there are places that used FindOneId that I switched with this function that handles err != nil and host == nil differently. I prefer to follow convention but I can change it if that's still preferred.